File: android/gemcore/src/testFixtures/kotlin/com/gemwallet/android/testkit/AssetDataMock.kt
package com.gemwallet.android.testkit

import com.gemwallet.android.model.AssetBalance
import com.gemwallet.android.model.AssetData
import com.gemwallet.android.model.AssetPriceInfo
import com.wallet.core.primitives.AssetMetaData

fun mockAssetData(balance: AssetBalance = AssetBalance.create(mockAsset()), price: AssetPriceInfo? = null, metadata: AssetMetaData = mockAssetMetaData()): AssetData {
    val asset = mockAsset()
    val account = mockAccount(chain = asset.id.chain)
    return AssetData.from(
        assetInfo = mockAssetInfo(
            asset = asset,
            balance = balance,
            price = price,
            metadata = metadata,
        ),
        wallet = mockWallet(accounts = listOf(account)),
        account = account,
    )
}

File: core/crates/primitives/src/testkit/asset_data_mock.rs
use crate::{Account, Asset, AssetData, AssetMetaData, Balance};

impl AssetData {
    pub fn mock(asset: Asset, balance: Balance) -> Self {
        Self {
            account: Account::mock(asset.chain(), ""),
            asset,
            balance,
            price: None,
            price_alerts: vec![],
            metadata: AssetMetaData::mock(),
            associations: vec![],
        }
    }
}
